File: app/models/week.rb
class Week < ApplicationRecord
  after_create :add_days
  validates :number, numericality: true

  belongs_to :phase
  belongs_to :plan, touch: true

  has_many :days
  has_many :workouts, through: :days

  private
    def add_days
      Day.create([
        { week: self, name: "Sunday" },
        { week: self, name: "Monday" },
        { week: self, name: "Tuesday" },
        { week: self, name: "Wednesday" },
        { week: self, name: "Thursday" },
        { week: self, name: "Friday" },
        { week: self, name: "Saturday" }
        ])
    end
end

File: app/javascript/react/containers/Week.js
import React from 'react';
import Day from './Day'

const Week = props => {

  let days = props.days.map(day => {

    return (
      <Day
        key = {day.id}
        id = {day.id}
        workouts = {day.workouts}
      />
    )
  })

  return(
    <div className="week row">
      {days}
    </div>
  )
};

export default Week;
